README.md

```diff
@@ -271,6 +195,43 @@ When we create a new `person` fixture using the two custom generators we get the
 }
 ```
 
+:::
+
+### Create Your Own Transformer
+
+Instead of using the opinionated `Fixture` class, you can extend the unopinionated `Transformer` and register the desired generators.
+
+```ts
+import { Transformer } from 'zod-fixture';
+
+const transform = new Transformer().extend([
+	/* insert your generators here */
+]);
+
+const value = transform.from(zodSchema);
+```
+
+## API
+
+### Generators
+
+To generate a value based on a zod type we're using what we call a `Generator`.
+
+To make your own generators simpler this library also includes some useful utility methods to generate data.
+
+In the example below we create our own `totalVisitsGenerator` to return more realastic numbers using the `random` utilities.
+
+```ts [input.ts]
```

Comment:
My idea here is to have the getting started with inputs and outputs, and the API section just small code snippets.